File: LeetCode Solutions/May LeetCoding Challenges/Maximum Gap.py
class Solution:
    def maximumGap(self, nums: List[int]) -> int:
        if(len(nums) <= 1):
            return 0
        nums.sort()
        maximumDifference = 0
        for num in range(1, len(nums)):
            maximumDifference = max(maximumDifference, nums[num]-nums[num - 1])
        return maximumDifference

File: LeetCode Solutions/May 2021 LeetCoding Challenge/Maximum Gap/Maximum Gap.cpp
class Solution {
public:
    int maximumGap(vector<int>& nums) {
        if(nums.size() <= 1){
            return 0;
        }
        sort(nums.begin(), nums.end());
        int maximumDifference = 0;
        for(int num=1; num<nums.size(); num++){
            maximumDifference = max(maximumDifference, nums[num]-nums[num - 1]);
        }
        return maximumDifference;
    }
};